tests/v2/test_waku_rln_relay.nim

```diff
@@ -732,4 +750,141 @@ suite "Waku rln relay":
       # check that the correct number of key pairs is created
       groupKeyPairs.len == StaticGroupSize
       # compare the calculated root against the correct root
-      root == STATIC_GROUP_MERKLE_ROOT
\ No newline at end of file
+      root == STATIC_GROUP_MERKLE_ROOT
+  
+  test "NonSpamProof Protobuf encode/init test":
+    var 
+      proof: ZKSNARK
+      merkleRoot: MerkleNode
+      epoch: Epoch
+      shareX: MerkleNode
+      shareY: MerkleNode
+      nullifier: Nullifier
+    # populate fields with dummy values
+    for x in proof.mitems : x = 1
+    for x in merkleRoot.mitems : x = 2
+    for x in epoch.mitems : x = 3
+    for x in shareX.mitems : x = 4
+    for x in shareY.mitems : x = 5
+    for x in nullifier.mitems : x = 6
+    
+    let 
+      nsp = NonSpamProof(proof: proof,
+                          merkleRoot: merkleRoot,
+                          epoch: epoch,
+                          shareX: shareX,
+                          shareY: shareY,
+                          nullifier: nullifier)
+      protobuf = nsp.encode()
+      decodednsp = NonSpamProof.init(protobuf.buffer)
+
+    check:
+      decodednsp.isErr == false
+      decodednsp.value == nsp
+
+  test "test proofVrfy and proofGen for a valid proof":
```

Comment:
proofVerify? Vrfy seems like a non-standard abbreviation :P

Reply:
It is very common in crypto papers :D (but I am going to change it)

Reply:
oh til